.github/workflows/test.yml

```diff
@@ -0,0 +1,29 @@
+name: Test
+
+on:
+  pull_request:
+  push:
+  release:
+    types: published
+
+jobs:
+  build:
+    runs-on: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install dependencies
+      run: |
+        sudo apt install -y python3-pytest
+        curl -L -O https://github.com/Kitware/CMake/releases/download/v3.16.2/cmake-3.16.2-Linux-x86_64.sh
+        chmod +x cmake-3.16.2-Linux-x86_64.sh
+        ./cmake-3.16.2-Linux-x86_64.sh --skip-license --prefix=/usr/local
+    - name: Build ninja
+      shell: bash
+      run: |
+        cmake -Bbuild-cmake -H. -DCMAKE_BUILD_TYPE=Debug -GNinja
+        cd build-cmake
+        ninja
+    - name: Unit tests
+      run: ./build-cmake/ninja_test
```

Comment:
to do this using CMake, consider

```sh
cd build-cmake
ctest
```

Reply:
What's the advantage though?

Reply:
The advantage is, besides not having to remember which test names to type, is that under certain conditions, especially on Windows, a test may silently fail. CTest ensures the failure is noisy on the console:

```
ninja\build> .\ninja_test.exe

ninja\build> ctest
Test project C:/Users/micha/code_other/ninja/build
    Start 1: NinjaTest
1/1 Test #1: NinjaTest ........................Exit code 0xc0000139
***Exception:   0.01 sec
```

Reply:
So `.\ninja_test.exe` did fail? GitHub Actions checks for exit code != 0.

Reply:
This would be to show that there wasn't some test unintentionally excluded in the CI. For example, if someone added a test in CMakeLists but didn't check that the CI script also explicitly calls that test. If CTest is used in a CMake CI script, then one knows all intended tests are run.

It's just a general "best practice" when using CMake, for most situations

Reply:
https://github.com/ninja-build/ninja/runs/673809404?check_suite_focus=true showed that ctest doesn't give enough output in some cases, while calling the test binary directly does: https://github.com/ninja-build/ninja/pull/1588/checks?check_run_id=673891970
I'd also want the output, even if the tests succeed.

Given that we only have one test binary (and probably won't add another one), I'll pass on ctest.

Reply:
OK that's cool -- for ctest, it wants uppercase `ctest -V`  or `ctest -VV`

Reply:
Oh. Why didn't it complain?